internal/resource/url.go

```diff
@@ -298,7 +303,8 @@ func (f *Fetcher) fetchFromHTTP(u url.URL, dest io.Writer, opts FetchOptions) er
 		}
 	}
 
-	dataReader, status, ctxCancel, err := f.client.getReaderWithHeader(u.String(), headers)
+	opts.Headers = headers
```

Comment:
Updating the `opts` variable with the content of `headers` is a change to what was previously done. Right now we won't notice it as `decompressCopyHashAndVerify` doesn't interact with `opts.Headers` but if we were to change that later and start using the field we could get bit by this. Instead I think it'd be better to copy the `opts` object and modify the `Headers` field on the copy for `httpReaderWithHeader`.